index.js

```diff
@@ -157,6 +157,7 @@ module.exports = function push (db, source, options, callback) {
 
     if (options.watch) {
       var queue = async.queue(function (task, done) {
+        console.log(task.type + ': ' + task.filepath)
```

Comment:
Alright, I simplified the whole thing.

Note I also added the awayWriteFinish flag as a precaution.